docker/Dockerfile

```
@@ -0,0 +1,302 @@
+    TERM=xterm-256color \
+    LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8
+
+WORKDIR /fluster
+
+# Base dependencies
+RUN apt-get update && apt-get install -y \
+    git \
+    wget \
+    curl \
+    ca-certificates \
+    software-properties-common \
+    python3 \
+    python3-pip \
+    python3-venv \
+    pciutils \
+    && rm -rf /var/lib/apt/lists/*
+
+# VAAPI support, Intel and AMD drivers
+RUN apt-get update && apt-get install -y \
+    libva-dev \
+    libva-drm2 \
+    libva-x11-2 \
+    vainfo \
+    i965-va-driver \
+    mesa-va-drivers \
+    && (apt-get install -y intel-media-va-driver-non-free || \
+        apt-get install -y intel-media-va-driver || true) \
+    && rm -rf /var/lib/apt/lists/*
+
+# VAAPI driver for NVIDIA (Ubuntu 24.04+)
+RUN if [ "${UBUNTU_VERSION}" = "24.04" ]; then \
+        apt-get update && \
+        apt-get install -y nvidia-vaapi-driver && \
+        rm -rf /var/lib/apt/lists/*; \
+    fi
+
+# VDPAU support and GPU drivers
+RUN apt-get update && apt-get install -y \
+    libvdpau-dev \
+    libvdpau1 \
+    vdpauinfo \
+    libvdpau-va-gl1 \
+    mesa-vdpau-drivers \
+    && rm -rf /var/lib/apt/lists/*
+
+# NVIDIA VDPAU libraries (must match host driver version)
+RUN if [ -n "${NVIDIA_DRIVER_VERSION}" ]; then \
+        apt-get update && \
+        apt-get install -y \
+            libnvidia-decode-${NVIDIA_DRIVER_VERSION} \
+            libnvidia-encode-${NVIDIA_DRIVER_VERSION} && \
+        rm -rf /var/lib/apt/lists/*; \
+    fi
+
+# Intel QuickSync support via Media SDK and oneVPL
+RUN apt-get update && apt-get install -y \
+    libmfx1 \
+    libmfx-tools \
+    && (apt-get install -y libmfx-gen1.2 || true) \
+    && (apt-get install -y libvpl-dev libvpl2 || true) \
+    && rm -rf /var/lib/apt/lists/*
+
+# FFmpeg installation (3 methods: system, static, source)
+# Method "source": Build FFmpeg from source with all backends (VAAPI, VDPAU, QuickSync)
+# Method "static": Download pre-built static FFmpeg binary
+# Method "system": Install FFmpeg from system packages (default)
+RUN if [ "${FFMPEG_INSTALL_METHOD}" = "source" ]; then \
+      apt-get update -qq && apt-get install -y \
+        autoconf \
+        automake \
+        build-essential \
+        cmake \
+        git-core \
+        libass-dev \
+        libfreetype6-dev \
+        libgnutls28-dev \
+        libmp3lame-dev \
+        libsdl2-dev \
+        libtool \
+        libva-dev \
+        libvdpau-dev \
+        libvorbis-dev \
+        libxcb1-dev \
+        libxcb-shm0-dev \
+        libxcb-xfixes0-dev \
+        meson \
+        ninja-build \
+        pkg-config \
+        texinfo \
+        wget \
+        yasm \
+        zlib1g-dev \
+        nasm \
+        libx264-dev \
+        libx265-dev \
+        libnuma-dev \
+        libvpx-dev \
+        libfdk-aac-dev \
+        libopus-dev \
+        libmfx-dev \
+        libaom-dev \
+        libdav1d-dev \
+        libunistring-dev \
+      && rm -rf /var/lib/apt/lists/* && \
```

Just below this line add compilation of aom and dav1d. Works on Ubuntu 20.04

```suggestion
      && rm -rf /var/lib/apt/lists/* && \
      mkdir /tmp/aom && \
      cd /tmp/aom && \
      git -C aom pull 2> /dev/null || git clone --depth 1 https://aomedia.googlesource.com/aom && \
      mkdir -p aom_build && \
      cd aom_build && \
      cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX="/usr/local" -DENABLE_TESTS=OFF -DENABLE_NASM=on ../aom && \
      make && \
      make install && \
      cd / && rm -rf /tmp/aom && \
      mkdir /tmp/dav1d && \
      cd /tmp/dav1d && \
      git -C dav1d pull 2> /dev/null || git clone --depth 1 https://code.videolan.org/videolan/dav1d.git && \
      mkdir -p dav1d/build && \
      cd dav1d/build && \
      meson setup -Denable_tools=false -Denable_tests=false --default-library=static .. --prefix "/usr/local" --libdir="/usr/local/lib" && \
      ninja && \
      ninja install && \
      cd / && rm -rf /tmp/dav1d && \
```

---

Tested with 20.04 and 22.04

```
docker run --rm fluster gst-inspect-1.0 2>&1 | Select-String -Pattern "av1|dav1d|aom"

aom:  av1dec: AV1 Decoder
aom:  av1enc: AV1 Encoder
libav:  avdec_wmav1: libav Windows Media Audio 1 decoder
libav:  avenc_wmav1: libav Windows Media Audio 1 encoder
```